citrination_client/client.py

```diff
@@ -191,43 +191,43 @@ def predict(self, model_name, candidates):
     def _get_predict_url(self, model_name):
         return self.api_url + '/csv_to_models/' + model_name + '/predict'
 
-    def upload_file(self, file_path, data_set_id, root_path=None):
+    def upload(self, dataset_id, source_path, dest_path=None):
         """
-        Upload file to Citrination.
-
-        :param file_path: File path to upload.
-        :param data_set_id: The dataset id to upload the file to.
-        :return: Response object or return code if the file was not uploaded.
+        Upload a file, specifying source and dest paths a file (acts as the scp command)
+        :param source_path: The path to the file on the source host
+        :param dest_path: The path to the file where the contents of the upload will be written (on the dest host)
+        :return: The response object or a boolean False if the file was not uploaded
         """
-        if os.path.isdir(str(file_path)):
-            for path, subdirs, files in os.walk(file_path):
-                for name in files:
-                    if root_path:
-                       root = root_path
-                    else:
-                       root = str(file_path)
-                    success = self.upload_file(os.path.join(path, name), data_set_id, root)
-                    print(success)
-            message = {"message": "Upload of files in " + str(root) + " is complete."}
-            return json.dumps(message)
+        source_path = str(source_path)
+        if not dest_path:
+            dest_path = source_path
         else:
-            url = self._get_upload_url(data_set_id)
-            file_data = {"file_path": str(file_path)}
-            if root_path:
-                file_data["root_path"] = root_path
+            dest_path = str(dest_path)
 
+        if os.path.isdir(source_path):
+            for path, subdirs, files in os.walk(source_path):
+                for name in files:
+                    path_without_root_dir = path.split("/")[1:] + [name]
+                    current_dest_path = os.path.join(dest_path, *path_without_root_dir)
+                    current_source_path = os.path.join(path, name)
+                    success = self.upload(dataset_id, current_source_path, current_dest_path)
+            message = {"message": "Upload of files is complete."}
+            return json.dumps(message)
+        elif os.path.isfile(source_path):
+            url = self._get_upload_url(dataset_id)
+            file_data = { "file": { "dest_path": str(dest_path), "src_path": str(source_path)}}
```

Comment:
The service has been updated, but I made some changes that make the service backward compatible with the old body.